Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt

```diff
@@ -28,11 +29,14 @@ class ConfirmCheckInViewModel @AssistedInject constructor(
 ) : CWAViewModel() {
     private val traceLocation = MutableStateFlow(verifiedTraceLocation.traceLocation)
     private val createJournalEntry = MutableStateFlow(true)
-    private val checkInLength = MutableStateFlow(
-        Duration.standardMinutes(
-            verifiedTraceLocation.traceLocation.defaultCheckInLengthInMinutes?.toLong() ?: 0L
-        )
+
+    private val truncatedDefaultCheckInLength = roundToNearestValidDuration(
+        verifiedTraceLocation.traceLocation.defaultCheckInLengthInMinutes ?: 0
     )
+    private val checkInLength = MutableStateFlow(truncatedDefaultCheckInLength)
+
+    private fun roundToNearestValidDuration(minutes: Int): Duration =
```

Comment:
Some tests would be nice :)

Reply:
I will use some of your logic in one of my upcoming PRs, so I could also add some tests for the rounding if you like.